src/psm/src/pdnsim.cpp

```diff
@@ -135,10 +163,10 @@ void PDNSim::analyzePowerGrid(const std::string& voltage_file,
   auto irsolve_h = getIRSolver(enable_em);
 
   if (!irsolve_h->build(error_file)) {
+    saveIRDrop(irsolve_h.get());
     logger_->error(
         utl::PSM, 78, "IR drop setup failed.  Analysis can't proceed.");
```

Comment:
Why do we save the results on failure?

Reply:
the GUI debugger relies on this to be saved. At present, we can only debug PSM when it is successful. This ensures the information about the nodes is copied over so we can debug it when it "claims" the PDN is not connected